### PR TITLE
test (relay-sampling): Add test to demonstrate more complex example

### PR DESCRIPTION
**Pull request by andriisoldatenko**
This PR add test to emulate what we have right now at Sentry, and it demonstrates that we want to put factor rule to the top.

#skip-changelog

### CODE REVIEW

**Review by iambriccardo — APPROVED** [review]

File: relay-sampling/src/lib.rs [hunk 1]
```diff
@@ -2770,6 +2758,100 @@ mod tests {
...
     }
 
+    #[test]
+    /// Tests that the multi-matching works for a mixture of trace and transaction
+    /// 1. Rule is healthcheck bias
+    /// 2. Rule is dev env bias
+    /// 3. Tx rebasing rule part a
+    /// 4. Tx rebasing rule part b
+    /// 5. Factor bias ( to fix rate )
+    /// 6. Uniform sample rate
```
Comment (olksdr) [inline relay-sampling/src/lib.rs]:
```suggestion
    /// Tests that the multi-matching works for a mixture of trace and transaction.
    ///
    /// 1. Rule is health-check bias
    /// 2. Rule is dev env bias
    /// 3. Tx rebasing rule part a
    /// 4. Tx rebasing rule part b
    /// 5. Factor bias ( to fix rate )
    /// 6. Uniform sample rate
```

File: relay-sampling/src/lib.rs [hunk 2]
```diff
@@ -2770,6 +2758,100 @@ mod tests {
...
+    fn test_match_against_rules_with_multiple_event_types_and_matches() {
+        let healthcheck_rule_id = 1002;
```
Comment (olksdr) [inline relay-sampling/src/lib.rs]: This `healthcheck_rule_id` var is used in one place, it's fine to hardcode this number in the rule id below (especially in tests).

**Review by olksdr — APPROVED** [review]
:shipit:

File: relay-sampling/src/lib.rs (hunk 2 above)
Comment (andriisoldatenko) [reply relay-sampling/src/lib.rs]: @olksdr thanks! fixed in 82c7da46c

**Pull request merged**